src/pages/en/core-concepts/framework-components.md

```diff
@@ -0,0 +1,155 @@
+---
+layout: ~/layouts/MainLayout.astro
+title: Framework Components
+description: Learn how to use React, Svelte, etc.
+---
+Astro allows you to build your site using JavaScript UI components from a variety of popular frameworks. Astro will run your [React](https://reactjs.org/), [Preact](https://preactjs.com/), [Svelte](https://svelte.dev/), [Vue](https://vuejs.org/), [SolidJS](https://www.solidjs.com/), [AlpineJS](https://alpinejs.dev/) and [Lit](https://lit.dev/) components at build time, then render your entire site to static HTML by default, removing all the JavaScript from the final page!
+
+```astro
+---
+import MyReactComponent from '../components/MyReactComponent.jsx';
+---
+<html>
+  <head>
+    <title>Look How Easy it is to Use a React Component!</title>
+  </head>
+  <body>
+    <h1>My Astro Page</h1>
+    <MyReactComponent />
+  </body>
+</html>
+```
+
+You can use any of these frameworks as templating languages to build static pages, taking advantage of the syntax and tooling you already know and love. This allows you to build with components you're familiar with, maybe even the exact same ones you've already written! And, you won't send any unnecessary JavaScript to the client to render static elements, even dynamically generated ones.
+
+For client-side interactivity, Astro allows you to opt-in to load a single component's JavaScript and dpendencies only when and where that component is used.
+
+> 📚 Read more about Astro's [partial hydration](/en/core-concepts/partial-hydration)
+
+## Configure Framework Renderers
+
+Your Astro configuration must include the renderer for each framework you use. You can find, and manually edit, your project's current list of available renderers in `astro.config.mjs`.
+
+```js
+export default ({
+// No renderers are needed for AlpineJS support, just use Astro components!
+renderers: [
+    '@astrojs/renderer-svelte',
+    '@astrojs/renderer-vue',
+    '@astrojs/renderer-react',
+    '@astrojs/renderer-preact',
+    '@astrojs/renderer-lit',
+    '@astrojs/renderer-solid',
+  ],
+
+});
+```
+
+⚙️ See the [Renderer Reference](/en/reference/renderer-reference) to learn about configuring custom renderers.
+
+## Build With Components
+
+Use your JavaScript framework components in your Astro pages, layouts and components just like Astro components! All your components can live together in `/src/components`, or can be organized in any way you like.
+
+### Render on a Page
```

Comment:
unnecessary sub-header here, safe to remove?

Also, is this section repeating the page overview at the top? If so, I'd trim the overview and remove everything after the example in the intro section (like the paragraph "You can use any of these frameworks as templating languages...") and merging that content down to this section. No need for the intro to be so long if we'll be covering the same content down here!

Reply:
resolved in "freds changes"